File: insertion_sort.cpp
void insertion_sort(int arr[], int size) {
	for (int i = 1; i < size; i++) {
		int temp = arr[i];
		int j = i - 1;
		for (; j >= 0; j--) {
			if (arr[j] > temp) {
				arr[j + 1] = arr[j];
			}
			else {
				break;
			}
		}
		arr[j + 1] = temp;
	}
}

void print(int arr[],int size) {
	for (int i = 0; i < size; i++) {
		cout << arr[i] << endl;
	}
}

int main() {
	int size = 10;
	int arr[10] = { 5,7,3,1,5,17,9,8,6,20 };
	insertion_sort(arr, 10);
	print(arr, size);
}

File: InsertionSort.java
public class InsertionSort {
    static void isort(int array1[],int n){
        for(int i=0;i < n;i++){
            for(int j=i;j > 0 && array1[j-1]>array1[j];j--){
                int temp = array1[j];
                    array1[j] = array1[j-1];
                    array1[j-1] = temp;
            }
        }

        for(int i=0;i<5;i++){
            System.out.print(array1[i] + " ");
        }
    }
    
    
    
    public static void main(String[] args) {
        int array1[] = {5,3,1,7,8};
        int size = array1.length;
        isort(array1,size);
    }
}
